src/main/java/training/busboard/models/Bus.java

```diff
@@ -0,0 +1,59 @@
+package training.busboard.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Bus {
+    private static Logger LOGGER = LogManager.getLogger();
+    private String lineId;
+    private int secondsUntilArrival;
+    private String stopName;
+
+    public Bus(String lineId, int secondsUntilArrival, String stopName) {
+        this.lineId = lineId.toUpperCase();
+        this.secondsUntilArrival = secondsUntilArrival;
+        this.stopName = stopName;
+    }
+
+    public int arrivesBefore(Bus otherBus) {
+        if (this.secondsUntilArrival < otherBus.secondsUntilArrival) {
+            return -1;
+        } else if (this.secondsUntilArrival > otherBus.secondsUntilArrival) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    public String toString() {
```

Comment:
I usually think that this sort of thing doesn't really belong in the `toString` method of a model. 
The idea of `toString` is that it is simply a convenient string representation of the data held by the class. But your `toString` is coupled tightly to your use-case for the model - this method couldn't be re-used if you also wanted to deal with buses in other contexts.

`toString` is mostly helpful for development and debugging. If you want use-case-specific representations of data, then that feels like it fits better in the user-facing classes, rather than the raw model.